CHANGELOG.md

```diff
@@ -10,6 +10,7 @@
 - Fish is now more forgiving of missing or invalid $TERM values (#3850).
 - The `string` command now supports a `repeat` subcommand with the obvious behavior (#3864).
 - The `string match` command now supports a `--filter` flag to emit the entire string partially matched by a pattern (#3957).
+- The `string replace` command now supports a `--filter` flag to limit output to strings which underwent a replacement (#3348).
```

Comment:
Okay, I'll change the `string match --filter` flag to `--full` in a separate commit.